roles/redpanda_broker/tasks/start-redpanda.yml

```diff
@@ -28,43 +28,48 @@
     loop_var: custom_config_template
   when: custom_config_template.condition | default(True)
 
-# Print the redpanda variable when -vvv is specified - hint, this is often malformed due to the difficulty in passing it in on the command line
-- name: Debug
+- name: Debug redpanda variable
   ansible.builtin.debug:
     var: redpanda
     verbosity: 3
   when: redpanda is defined
 
-- name: Debug
-  ansible.builtin.debug:
-    msg: "{{ redpanda | from_json }}"
-    verbosity: 3
+- name: Parse redpanda variable
+  ansible.builtin.set_fact:
+    parsed_redpanda: "{{ redpanda if (redpanda is mapping) else (redpanda | from_json) }}"
   when: redpanda is defined
 
-# merge in user's overrides of items in the templates
-- name: Merge with user-provided overrides (via redpanda variable)
+- name: Set default empty dict if redpanda is not defined
   ansible.builtin.set_fact:
-    configuration: "{{ configuration | combine((redpanda | default('{}') | from_json), recursive=True) }}"
+    parsed_redpanda: {}
+  when: redpanda is not defined
 
-# Print the host_specific_override variable when -vvv is specified - hint, this is often malformed due to the difficulty in passing it in on the command line
-- name: Debug
+- name: Debug parsed redpanda
+  ansible.builtin.debug:
+    var: parsed_redpanda
+
+- name: Debug host_specific_override variable
   ansible.builtin.debug:
     var: host_specific_override
-    verbosity: 3
-  when: redpanda is defined
+  when: host_specific_override is defined
 
-- name: Debug
+- name: Set default empty dict if host_specific_override is not defined
+  ansible.builtin.set_fact:
+    host_specific_override: {}
+  when: host_specific_override is not defined
+
+- name: Debug parsed host_specific_override
   ansible.builtin.debug:
-    msg: "{{ host_specific_override | from_json }}"
-    verbosity: 3
-  when: redpanda is defined
+    var: host_specific_override
 
-# merge in user's overrides of items in the templates
-- name: Merge with user-provided host specific overrides (via host_specific_override variable)
+- name: Merge with user-provided overrides (via redpanda variable)
   ansible.builtin.set_fact:
-    configuration: "{{ configuration | combine((host_specific_override | default('{}') | from_json), recursive=True) }}"
+    configuration: "{{ configuration | combine(parsed_redpanda, recursive=True) }}"
 
-# on first startup, the redpanda process will create this directory and
+- name: Merge with user-provided host specific overrides
+  ansible.builtin.set_fact:
+    configuration: "{{ configuration | combine(host_specific_override, recursive=True) }}"
+    # on first startup, the redpanda process will create this directory and
```

Comment:
Suggesting the following. There's a decent chance for typos and errors in each host override and because of the way error handling works you could end up with one host failing and then dropping out of the run. In general we don't really have that problem in the rest of the role because everything is done cluster-wide (maybe other than the rolling restart logic). But here I think it warrants having this safety check.
```suggestion
    configuration: "{{ configuration | combine(host_specific_override, recursive=True) }}"
  any_errors_fatal: true
# on first startup, the redpanda process will create this directory and
```